worksheet.rb

```diff
@@ -1,34 +1,150 @@
-# Which layers are "next" to each other?
-
-########################################################
-# Step 2: Assign a data structure to each layer
-
-# Copy your list from above, and in this section
-# determine what data structure each layer should have
-
-########################################################
-# Step 3: Make the data structure!
-
-# Setup the entire data structure:
-# based off of the notes you have above, create the
-# and manually write in data presented in rides.csv
-# You should be copying and pasting the literal data
-# into this data structure, such as "DR0004"
-# and "3rd Feb 2016" and "RD0022"
-
-########################################################
-# Step 4: Total Driver's Earnings and Number of Rides
-
-# Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+driver_trips = [
+  {DR0001: [{"3rd Feb 2016": [{cost: 10, rider_id: "RD0003", rating: 3},
+                              {cost: 30, rider_id: "RD0015", rating: 4}]},
+            {"5th Feb 2016": [{cost: 45, rider_id: "RD0003", rating: 2}]}]},
+  {DR0002: [{"3rd Feb 2016": [{cost: 25, rider_id: "RD0073", rating: 5}]},
+            {"4th Feb 2016": [{cost: 15, rider_id: "RD0013", rating: 1}]},
+            {"5th Feb 2016": [{cost: 35, rider_id: "RD0066", rating: 3}]}]},
+  {DR0003: [{"4th Feb 2016": [{cost: 5, rider_id: "RD0066", rating: 5}]},
+            {"5th Feb 2016": [{cost: 50, rider_id: "RD0003", rating: 2}]}]},
+  {DR0004: [{"3rd Feb 2016": [{cost: 5, rider_id: "RD0022", rating: 5}]},
+            {"4th Feb 2016": [{cost: 10, rider_id: "RD0022", rating: 4}]},
+            {"5th Feb 2016": [{cost: 20, rider_id: "RD0073", rating: 5}]}]}
+]
+
+def average_rating(array)
+  rating_hash = {}
+  array.each do |driver_hash|
+    rating_array = []
+    driver_hash.each do |driver, date_array|
+      date_array.each do |date_hash|
+        date_hash.each do |_date, ride_array|
+          ride_array.each do |ride_data|
+            rating_array.push(ride_data[:rating])
+          end
+        end
+      end
+      rating_hash[driver] = (rating_array.sum / rating_array.length.to_f).round(1)
+    end
+  end
+  return rating_hash
+end
+
+def nice_average(hash)
+  string = ""
+  hash.each do |key, value|
+    string += "\nDriver #{key}: #{value} rating"
+  end
+  return "Average ratings:" + string + "\n\n"
+end
+
+def total_rides(array)
+  ride_hash = {}
+  array.each do |driver_array|
+    rides = 0
+    driver_array.each do |driver, ride_array|
+      ride_array.each do |date_hash|
+        date_hash.each do |_date, ride_array|
+          rides += ride_array.count
+        end
+      end
+      ride_hash[driver] = rides
+    end
+  end
+  return ride_hash
+end
+
+def nice_rides(hash)
```

Comment:
Nice work putting the work of making "nice" output in a method. Consider giving your parameters more meaningful names to increase readability.